textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/nlp/sentiment/EmojiSentimentOperator.java

```diff
@@ -0,0 +1,198 @@
+package edu.uci.ics.textdb.exp.nlp.sentiment;
+
+import edu.uci.ics.textdb.api.constants.ErrorMessages;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.api.exception.DataFlowException;
+import edu.uci.ics.textdb.api.exception.TextDBException;
+import edu.uci.ics.textdb.api.field.IField;
+import edu.uci.ics.textdb.api.field.IntegerField;
+import edu.uci.ics.textdb.api.schema.Attribute;
+import edu.uci.ics.textdb.api.schema.AttributeType;
+import edu.uci.ics.textdb.api.schema.Schema;
+import edu.uci.ics.textdb.api.tuple.Tuple;
+import edu.uci.ics.textdb.api.utils.Utils;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This Operator performs sentiment analysis using Simple Emoji and Emoticon analysis targeted specifically for tweets
+ * using unicode, browser escape character and smiley regex matching.
+ *
+ * The result is an integer indicating the sentiment score, which represents:
+ *
+ * 3 - positive
+ * 2 - neutral
+ * 1 - negative
+ *
+ *
+ * The result will be put into an attribute with resultAttributeName specified in predicate, and type Integer.
+ *
+ * @author Vinay Bagade
+ *
+ */
+public class EmojiSentimentOperator implements IOperator {
+    private final EmojiSentimentPredicate predicate;
+    private IOperator inputOperator;
+    private Schema outputSchema;
+    private int cursor = CLOSED;
+    public static final Pattern SMILEY_REGEX_PATTERN = Pattern.compile(".*(:[)DdpP]|:[ -]\\)|<3)+.*");
```

Comment:
add some comments and try to explain these two regexes. Regex can look like magic if you don't explain them :)